raiden-dapp/src/components/dialogs/UdcDepositDialog.vue

```diff
@@ -90,32 +108,53 @@ import { RaidenError } from 'raiden-ts';
   },
 })
 export default class UdcDepositDialog extends Vue {
-  amount: string = '10';
+  mainnet!: boolean;
+  uniswapURL: string = '';
+  udcToken!: Token;
+  utilityTokenAmount: string = '';
+  step: 'mint' | 'approve' | 'deposit' | '' = '';
   loading: boolean = false;
   error: Error | RaidenError | null = null;
 
-  step: 'mint' | 'approve' | 'deposit' | '' = '';
-
   @Prop({ required: true, type: Boolean })
   visible!: boolean;
 
-  mainnet!: boolean;
-  udcToken!: Token;
+  @Emit()
+  cancel() {
+    this.error = null;
+  }
 
   get valid(): boolean {
-    return /^[1-9]\d*$/.test(this.amount);
+    return /^[1-9]\d*$/.test(this.utilityTokenAmount);
   }
 
-  @Emit()
-  cancel() {
-    this.error = null;
+  async mounted() {
```

Comment:
What was the reason here for you to go for the lifecycle hook? Most of the stuff done here looks to me like it could be sliced into plain computed values. Especially the difference between the two `utilityTokenAmount` cases would be more clear to see on such an approach.
The only thing I personally see here is that the `mainAccountUtilityTokenBalance()` call could take a while. If you're intention was to don't show anything until this has finished, I would consider to shrink this function for this part only and introduce a loading state for it. So the user will get the dialog but sees the spinner. Last is optional nice solution I simply suggest. 

PS: You won't need a condition for the URL value. Just calculate it for the test network as well. The template makes sure to only display it when needed. So no need to make this decision twice in the code.

Reply:
Regarding the ```utilityTokenAmount```. If I add it as a computed property, it will first be empty and then not update accordingly once it has a value. That will always leave the dialog with an empty string. If you look in the ```Withdrawal.vue``` component, for instance, we use the same approach. But perhaps I'm not seeing something here, could you provide an example of your idea?

Regarding the URL conditional, I changed the code a little bit when fixing the issue where the input also updated the available balance. I still need the conditional I believe, but have a look and let me know if I've could have done it in another way.

Reply:
Note: we had a call to discuss this PR.

We can outsource everything except the asynchronous token amount fetching. Also will the mainnet conditation not be necessary anymore.

Reply:
Unfortunately, #1983 needs to be further investigated befor these proposed changes can be made.